sdk/src/types/block/output/delegation.rs

```diff
@@ -248,22 +262,20 @@ impl DelegationOutput {
     /// Creates a new [`DelegationOutputBuilder`] with a provided amount.
     pub fn build_with_amount(
         amount: u64,
-        delegated_amount: u64,
```

Comment:
Honestly this is very confusing and I don't like it. If `delegated_amount` is the one you actually set, this function and all others should be called `build_with_delegated_amount` and the param should keep its name. I don't like that we are deviating from the TIP, I don't want to delay this PR even further but I think I would like to change that back in the future.

Reply:
\>:c